File: python/1423. 可获得的最大点数.py
from typing import List
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        res = cur = sum(cardPoints[:k])
        for i in range(k):
            cur += cardPoints[-i - 1] - cardPoints[k - 1 - i]
            res = max(cur, res)
        return res

s = Solution()
cardPoints = [1,2,3,4,5,6,1]
k = 3
print(s.maxScore(cardPoints,k))

File: go/general/1423. 可获得的最大点数.go
package main

func sum(a []int) int {
	res := 0
	for _, i := range a {
		res += i
	}
	return res
}

func max(a, b int) int {
	if a > b {
		return a
	} else {
		return b
	}

}

func maxScore(cardPoints []int, k int) int {
	res := sum(cardPoints)
	cur := res
	for i, _ := range cardPoints {
		cur += cardPoints[len(cardPoints)-1-i] - cardPoints[k-1-i]
		res = max(cur, res)
	}
	return res
}
